testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestHiveStorageFormats.java

```diff
@@ -542,6 +548,92 @@ public void testOrcTableCreatedInTrino()
         onTrino().executeQuery("DROP TABLE orc_table_created_in_trino");
     }
 
+    @Test(dataProvider = "storageFormatsWithConfiguration", groups = STORAGE_FORMATS_DETAILED)
+    public void testInsertAllDataTypesWithTrino(StorageFormat storageFormat)
+    {
+        // only admin user is allowed to change session properties
+        setAdminRole();
+        setSessionProperties(storageFormat);
+
+        String tableName = "storage_formats_compatibility_data_types_" + storageFormat.getName().toLowerCase(ENGLISH);
+
+        onTrino().executeQuery(format("DROP TABLE IF EXISTS %s", tableName));
+
+        StringBuilder createTableBuilder = new StringBuilder();
+        createTableBuilder.append(format("CREATE TABLE %s", tableName));
+        List<String> columns = new ArrayList<>();
+        if (!"AVRO".equals(storageFormat.name)) {
+            columns.add("c_tinyint           TINYINT");
+            columns.add("c_smallint          SMALLINT");
+        }
+        columns.add("c_int               INTEGER");
+        columns.add("c_bigint            BIGINT");
+        columns.add("c_float             REAL");
+        columns.add("c_double            DOUBLE");
+        columns.add("c_decimal           DECIMAL(10,0)");
+        columns.add("c_decimal_w_params  DECIMAL(10,5)");
+        columns.add("c_timestamp         TIMESTAMP(3)");
+        columns.add("c_date              DATE");
+        columns.add("c_string            VARCHAR");
+        columns.add("c_varchar           VARCHAR(10)");
+        columns.add("c_char              CHAR(10)");
+        columns.add("c_boolean           BOOLEAN");
+        columns.add("c_binary            VARBINARY");
```

Comment:
Are there any other types that should be included here for testing?

Reply:
`time` seems not to be supported in `io.trino.plugin.hive.util.HiveTypeTranslator#toTypeInfo`

Reply:
correct & expected, per my comment about applicability

Reply:
have noted them down here for completeness, since i strive to apply the same ordering  in other places too